src/components/ListView/ListView.js

```diff
@@ -0,0 +1,180 @@
+    location: PropTypes.shape({ search: PropTypes.string.isRequired }).isRequired,
+    match: PropTypes.shape({ path: PropTypes.string.isRequired }).isRequired,
+    history: PropTypes.shape({ push: PropTypes.func.isRequired }).isRequired,
+    label: PropTypes.node.isRequired,
+    selectedEntity: PropTypes.object.isRequired,
+    checkboxList: checkboxListShape,
+    setList: PropTypes.func.isRequired,
+    showSingleResult: PropTypes.bool,
+    EntityView: PropTypes.func.isRequired,
+    EntityForm: PropTypes.func.isRequired,
+    INITIAL_RESULT_COUNT: PropTypes.number,
+    RESULT_COUNT_INCREMENT: PropTypes.number,
+    ENTITY_KEY: PropTypes.string,
+    actionMenuItems: PropTypes.arrayOf(PropTypes.string).isRequired,
+    visibleColumns: PropTypes.arrayOf(PropTypes.string).isRequired,
+    columnWidths: PropTypes.PropTypes.object.isRequired,
+    initialValues: PropTypes.object.isRequired,
+    renderHeaders: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    showSingleResult: true,
+    checkboxList: {
+      selectedRecords: new Set(),
+      isAllSelected: false,
+      selectRecord: noop,
+      deselectAll: noop,
+      handleSelectAllCheckbox: noop,
+    },
+    // big numbers to get rid of infinite scroll
+    INITIAL_RESULT_COUNT: 5000,
+    RESULT_COUNT_INCREMENT: 5000,
+    ENTITY_KEY: '',
+  };
+
+  componentDidMount() {
+    this.setList();
+  }
+
+  componentDidUpdate(prevProps) {
+    const { resources } = this.props;
+
+    if (prevProps.resources !== resources) {
+      this.setList();
+    }
+  }
+
+  get entityList() {
+    const { ENTITY_KEY } = this.props;
+
+    return get(this.props.resources, [ENTITY_KEY, 'records'], []);
+  }
+
+  setList() {
+    const { setList } = this.props;
+
+    setList(this.entityList);
+  }
+
+  renderActionMenu = menu => (
+    <ActionMenu
+      entity={this}
+      menu={menu}
+    />
+  );
+
+  render() {
+    console.log('Props resulted: ', this.props);
+
+    const {
+      resources,
+      mutator,
+      location: { search },
+      label,
+      location,
+      history,
+      match,
+      showSingleResult,
+      selectedEntity,
+      checkboxList: {
+        selectedRecords,
+        isAllSelected,
+        selectRecord,
+        deselectAll,
+        handleSelectAllCheckbox,
+      },
+      INITIAL_RESULT_COUNT,
+      RESULT_COUNT_INCREMENT,
+      initialValues,
+      ENTITY_KEY,
+      visibleColumns,
+      columnWidths,
+      EntityView,
+      EntityForm,
+      renderHeaders,
+    } = this.props;
+
+    const urlQuery = queryString.parse(search);
+    const searchTerm = trimSearchTerm(urlQuery.query);
+
+    return (
+      <IntlConsumer>
+        {intl => (
+          <ViewContainer
+            entityKey={ENTITY_KEY}
+            mutator={mutator}
+            location={location}
+            history={history}
+            match={match}
+            selectRecord={selectRecord}
+          >
+            {props => (
+              <Fragment>
```

Comment:
This `Fragment` is not needed here. please remove

Reply:
Done